Changes in Controller & Model files (messages & default state functionality)

priiish opened the pull request:
Pull Request for Issue # .

### Summary of Changes
**Issue1:** 
While managing states and trying to delete a state that is set to default the user received an error message saying "Item must be published to set default state." (COM_WORKFLOW_ITEM_MUST_PUBLISHED) whereas it should be "You are trying to delete the default item." (COM_WORKFLOW_MSG_DELETE_DEFAULT).
=> StateModel.php JText

Furthermore the user received a confirmation message even if 0 items were deleted. Changed condition of last if statement in function publish(). Now a confirmation message is only sent if the number of deleted items is greater than 0.
=> AdminController.php

**Issue2:**
It was not possible to change the default status of the states via the star icon. Fixed this issue.
=> StateModel.php

### Testing Instructions
-> Go to Content / Workflows / Workflows / Manage
- Try to delete a default state and take a look at the (error) message concerning the process and the number of deleted items.
- Try to change the default status of a state via the star icon (must be published).

### Expected result
Appropriate (error) messages and functionality of the default option (star icon).


### Actual result
Works as expected.

### Documentation Changes Required
None.

priiish committed:
StateModel.php: changed JText (enqueueMessage) due to fact, that wrong error was thrown. When trying to delete default state, user received message "Item must be published to set default state" whereas it must be "You are trying to delete the default item.".

StateModel.php: changed condition in first if statement in function setDefault().

AdminController.php: commented out else case due to it being unnecessary in the function publish().

priiish committed:
AdminController.php: reviewed if statement, fixed case. Changed condition in last if statement. Now confirmation message is only sent in case of "real" deletion (>= 0).

priiish committed:
inserted missing "else" in case (cleanup)

priiish committed:
restored if condition because there was no need for a change => changed by mistake

priiish committed:
final version: the condition needs to be $table->published == 0 because otherwise the default selection via the star icon wont work #no mistake

priiish committed:
Better code in regard to condition because if not value 1 then item is
either trashed oder unpublished and must be published first (before
default can be set).

bembelimen commented on libraries/src/MVC/Controller/AdminController.php:
```diff
@@ -244,12 +244,12 @@ public function publish()
 				}
 
-				if ($ntext !== null)
+				if (count($cid) != 0)
```
I think you could use ```if (count($cid))``` because values larger to 0 equals to ```true```

bembelimen commented on administrator/components/com_workflow/Model/StateModel.php:
```diff
@@ -320,7 +320,8 @@ public function publish(&$pks, $value = 1)
 				{
 					// Prune items that you can't change.
-					$app->enqueueMessage(\JText::_('COM_WORKFLOW_ITEM_MUST_PUBLISHED'), 'error');
+					$app->enqueueMessage(\JText::_('COM_WORKFLOW_MSG_DELETE_DEFAULT'), 'error');
```
Does this language string exists?

priiish replied:
Yes ist does. Have a look at joomla-cms/administrator/language/en-GB/en-GB.com_workflow.ini line 51.

bembelimen commented on administrator/components/com_workflow/Model/StateModel.php:
```diff
@@ -265,7 +265,7 @@ public function setDefault($pk, $value = 1)
 		{
-			if ($table->published !== 1)
+			if ($table->published != 1)
 			{
```
Perhaps ```!$table->published```?

priiish committed:
Updated conditions in AdminController.php and StateModel.php

priiish committed:
Fixed _canDeleteCategories() and associated function to count items ->
table could not be accessed, therefore deletion was not possible
& error message was being thrown

priiish committed:
Created new event (event_before_delete_categories) so now categories that
have items cannot be moved to trash anymore.
